sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.throughputControl.ThroughputControlMode;
+import com.azure.cosmos.util.Beta;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * Group configuration which will be used in Throughput control.
+ */
+@Beta(value = Beta.SinceVersion.V4_10_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class ThroughputControlGroup {
+
+    private final static boolean DEFAULT_USE_BY_DEFAULT = false;
+    private final static ThroughputControlMode DEFAULT_CONTROL_MODE = ThroughputControlMode.LOCAL;
+
+    private ThroughputControlMode controlMode;
+    private String groupName;
+    private CosmosAsyncContainer targetContainer;
+    private Integer targetThroughput;
+    private Double targetThroughputThreshold;
+    private boolean useByDefault;
+
+    public ThroughputControlGroup() {
+        this.controlMode = DEFAULT_CONTROL_MODE;
+        this.useByDefault = DEFAULT_USE_BY_DEFAULT;
+    }
+
+    /**
+     * Get throughput group control mode.
+     *
+     * By default, it will be local control mode.
+     *
+     * @return the {@link ThroughputControlMode}.
+     */
+    public ThroughputControlMode getControlMode() {
+        return this.controlMode;
+    }
+
+    /**
+     * Set throughput group control mode to local.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup localControlMode() {
+        this.controlMode = ThroughputControlMode.LOCAL;
+        return this;
+    }
+
+    /**
+     * Set the throughput control group name.
+     *
+     * @param groupName
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup groupName(String groupName) {
+        checkArgument(StringUtils.isNotEmpty(groupName), "Group name can not be null or empty");
+        this.groupName = groupName;
+
+        return this;
+    }
+
+    /**
+     * Get the throughput control group name.
+     *
+     * @return the group name.
+     */
+    public String getGroupName() {
+        return this.groupName;
+    }
+
+    /**
+     * Get the throughput control group target container.
+     *
+     * @return the {@link CosmosAsyncContainer}.
+     */
+    public CosmosAsyncContainer getTargetContainer() {
+        return this.targetContainer;
+    }
+
+    /**
+     * Set the throughput control group target container.
+     *
+     * @param targetContainer the target container for the throughput control group.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    public ThroughputControlGroup targetContainer(CosmosAsyncContainer targetContainer) {
```

Comment:
Do we need a full CosmosAsyncContainer object here or database plus collection name/ID are enough?

Reply:
just to add a bit more context why this change request; as is, this class takes a dependency on an "async" entity which will make it much difficult to add support for it in the "sync" model.

Reply:
For local control mode, I think it can use either cosmosAsyncContainer or the name/id of the container. However, for distributed control, we will need the user to provide another container to keep tracking the traffic load, and this container can be in the same database or different or use different client. so for consistency, I thought this might look cleaner.  For distributed control, it will be something like: 
ThroughputControlGroupConfig group3 =
           new ThroughputControlGroupConfig()
                .groupName("group-3")
                .targetContainer(feedContainer)
               .targetThroughputThreshold(0.9)
               .distributedControlMode(
                    new DistributedThroughputControlConfig()
                    .controlContainer(throughputControlContainer) // can be under the same account as the feed container or different account
                   .documentRenewalInterval(Duration.ofSeconds(10))
                   .documentExpireInterval(Duration.ofSeconds(10)));

Reply:
I agree with Milis, if we can find a way to avoid mixing ThroughputControlGroup and CosmosAsyncContainer that would be great. for the same reasons Milis explained.

Reply:
another option is that we instantiate ThroughputControlGroup through CosmosAsyncContainer and CosmosContainer that way you can bind the target container to the throughput control group.

```
ThroughputControlGroup group = cosmosAsyncContainer.createThroughputGroup(String groupName);
```

Reply:
As long as we are all fine with also exposing a factory method to create the DistributedThroughputControlConfig form CosmosAsyncContainer I like the approach Mo proposed as well.

Reply:
yea, that looks better, and I think it is naturally fall into the hierarchy very well. 

Changed to use the proposed way: 
        ThroughputControlGroup group1 = container.createThroughputControlGroup("group-1");
        group1.setTargetThroughput(5) 
            .setLocalControlMode()
            .setUseByDefault();